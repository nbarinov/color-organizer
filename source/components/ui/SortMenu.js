import PropTypes from 'prop-types';

import '../../style/sort-menu.css';

const options = {
    date: 'SORTED_BY_DATE',
    title: 'SORTED_BY_TITLE',
    rating: 'SORTED_BY_RATING',
};

const SortMenu = ({ className, sort = 'SORTED_BY_DATE', onSelect = f => f }) => (
    <nav className={(className) ? className + ' sort-menu' : 'sort-menu'}>
        <h2 className="sort-menu__title">Sort Colors</h2>
        {Object.keys(options).map((item, i) =>
            <a key={i}
                href="#"
                className={(sort === options[item]) ? 'sort-menu__item sort-menu__item--selected' : 'sort-menu__item'}
                onClick={e => {
                    e.preventDefault();
                    onSelect(item);
                }} title={options[item]}>{item}</a>
        )}
    </nav>
);

SortMenu.propTypes = {
    className: PropTypes.string,
    sort: PropTypes.string,
    onSelect: PropTypes.func,
};

export default SortMenu;
