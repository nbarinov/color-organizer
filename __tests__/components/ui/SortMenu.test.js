import SortMenu from '../../../source/components/ui/SortMenu';

const { mount, shallow } = Enzyme;

describe("<SortMenu /> UI Component", () => {
    let wrapper;

    describe("Rendering for UI", () => {
        beforeAll(() => wrapper = mount(<SortMenu sort="SORTED_BY_TITLE" />));

        it("Renders Correctly", () => {
            expect(
                shallow(<SortMenu sort="SORTED_BY_TITLE" />).html()
            ).toMatchSnapshot()
        });

        it("renders parental className stars", () =>
            expect(
                shallow(<SortMenu className="parant__sort-menu" />)
                    .find('nav.parant__sort-menu.sort-menu')
                    .length
            ).toBe(1)
        );
    });

    describe("Changing the Sort", () => {
        it("click does not cause error", () => {
            mount(<SortMenu sort="SORTED_BY_TITLE" />)
                .children()
                .childAt(2)
                .simulate('click');
        });

        it("onSelect returns correct option", () => {
            const _select = jest.fn();
            const wrapper = mount(<SortMenu onSelect={_select} />);

            wrapper.children().childAt(3).simulate('click');

            expect(_select.mock.calls[0][0]).toBe("rating");
        });
    });
})