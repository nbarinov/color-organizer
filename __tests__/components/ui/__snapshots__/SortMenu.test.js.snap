// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SortMenu /> UI Component Rendering for UI Renders Correctly 1`] = `"<nav class=\\"sort-menu\\"><h2 class=\\"sort-menu__title\\">Sort Colors</h2><a href=\\"#\\" class=\\"sort-menu__item\\" title=\\"SORTED_BY_DATE\\">date</a><a href=\\"#\\" class=\\"sort-menu__item sort-menu__item--selected\\" title=\\"SORTED_BY_TITLE\\">title</a><a href=\\"#\\" class=\\"sort-menu__item\\" title=\\"SORTED_BY_RATING\\">rating</a></nav>"`;
